style-guide.md

```diff
@@ -0,0 +1,112 @@
+	* If working with/installing something requiring root permissions such as Docker, mention that root permission is needed once, then assume from that point forward that your HPC users have root privileges
+	* However, do not assume that HPC users know how programs you introduce may work differently on an HPC
+		* Example: Running a WDL on an HPC via Cromwell (ie via the Dockstore CLI) requires additional runtime arguments that a local user does not need to include in their WDL program, so make sure to either include those runtime arguments in your examples or tell HPC users what additional things they need
+* Assume Linux users are using Ubuntu or know how to translate into their own distro
+	* Exception: If directing very new users to install software via apt-get, consider mentioning that the instructions are specific to Ubuntu -- UCSC's linux lab for students runs on CentOS (which uses yum instead) so this is not a rare occurrence
+* Keep in mind that some of our users are using Macs, so make a note about these situations in the text if they occur:
+	* If there is an important functional difference between the Linux and the Mac version, such as xargs
+	* If there is an important installation difference between the Linux and the Mac version, such as the installation of the Dockstore CLI
+	* If something does not work/works differently on Apple Silicon Macs
+	* If there is an important difference between the bash and zsh version of something
+		* Do not assume that a Mac user is aware of whether they are using bash or zsh (the default changed in Catalina) or knows how to change between the two
+		* Do not worry about other shells
+
+## Coding in RST
+* Write in RST, not markdown
+* Use `.. comment` to write comments
+* Use consistent indentation spacing within a single RST file
+* Do not break up body text with a newline every x characters
+* Use use \`this style \<https://example.com\>_` of formatting external hyperlinks, as it is less error prone than the method involving setting the target in another paragraph
+* Use headings in this way throughout the entire repository to avoid issues with embedding RST files into other RST files:
+
+> `page title`  
+> `==========`
+> 
+> `heading level 1`  
+> `---------------`
+>
+> `heading level 2`  
+> `~~~~~~~~~~~~~~~`
+>
+> `heading level 3`  
+> `***************`
+>
+> `heading level 4`  
+> `+++++++++++++++`
+>
+> `heading level 5`  
+> `\``````````````
+>
+ 
+## International
+See also: https://developers.google.com/style/translation 
+
+* Avoid using idioms ("it's raining cats and dogs in Santa Cruz")
+* Avoid using culturally-specific comparisons ("the button is green like a dollar bill")
+* Try to use relatively simple English whenever possible
+* Make sure programs/templates you are providing to the user have a way to handle non-ASCII names, family-name-first ordering, and names with apostrophes and hyphens
```

Comment:
I don't understand what you mean by "family-name-first ordering" in this context.

Reply:
I mean what is often called "Eastern name order." For example, the current PM of Japan inherited the name Kishida from his father, in the same way I inherited the name O'Farrell from my father. The Wikipedia article about the PM of Japan is titled Fumio Kishida in English because most English-speakers refer to people by their personal name (Fumio/Ash) first and their family name (Kishida/O'Farrell) last. But his article in Japanese would be titled 岸田 文雄, which is Kishida Fumio, because in Japan people's family names come first.

Taking a step back and talking about the who bullet point: I was basically thinking we should avoid examples that break upon apostrophes since that is a thing that happens to me every so often, and when thinking about other things that might break, I maybe got a little carried away by also including naming order. Naming order is admittedly unlikely to come up in documentation -- honestly, this line might fit better in a style guide for user-facing code then docs -- but there is a chance of confusion if our docs use words like "first name" and "last name" instead of "given/personal name" and "family name."

Reply:
As an example of what I mean r/e apostrophe examples:

We shouldn't have something like this as an example for beginning users:
`string wdl_name_to_welcome = 'Julian Lucas'`

Because if you have a name like mine, things will break, and if you are a beginning user you may not understand why.
`string wdl_name_to_welcome = 'Ash O'Farrell'`

Instead, by following the style guide r/e names with apostrophes and hyphens, we'd write:
`string wdl_name_to_welcome = "Julian Lucas"`

Reply:
I am somewhat familiar with family name ordering differences in different parts of the world, but I was confused how this text applies to Dockstore documentation guidelines. Unless we have sample programs sorting names, I still don't see the need for "family-first ordering".

On using the term "family name" instead of "last name", makes sense in general, and I guess it doesn't hurt, but I don't think the doc is using any of those terms, and it's unlikely it will.

Reply:
Previously I was thinking how we handle the name field in dockstore.yml files may apply to this, but that isn't really a documentation thing. I'll take out the name ordering part then.